test/gtest/ucp/test_ucp_tag_fallback.cc

```diff
@@ -30,30 +30,28 @@ class test_ucp_tag_fallback : public ucp_test {
 
 UCS_TEST_P(test_ucp_tag_fallback, fallback)
 {
-    void *send_buffer;
-    void *recv_buffer;
-    ucs_status_ptr_t send_req;
-    ucs_status_ptr_t recv_req;
     ucp_request_param_t param = {0};
 
-    /* allocate read-only pages - it force to fail ibv_reg_mr calls */
-    send_buffer = mmap(0, MSG_SIZE, PROT_READ, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
+    /* allocate read-only pages - it force ibv_reg_mr() failure */
+    void *send_buffer = mmap(NULL, MSG_SIZE, PROT_READ,
+                             MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
     ASSERT_NE(send_buffer, MAP_FAILED);
 
-    recv_buffer = malloc(MSG_SIZE);
+    ucs::auto_ptr<char> recv_buffer(new char[MSG_SIZE]);
```

Comment:
auto_ptr knows to do delete[] instead of delete?

Reply:
no, but it should not be sensitive for char[]

Reply:
valgrind would bark

Reply:
let's wait for tests